src/jsonpatcherproxy.js

```diff
@@ -34,73 +39,89 @@ const JSONPatcherProxy = (function() {
   JSONPatcherProxy.escapePathComponent = escapePathComponent;
 
   /**
-   * Walk up the parenthood tree to get the path
+   * Walk up the tree metadata to get the path
    * @param {JSONPatcherProxy} instance 
    * @param {Object} tree the object you need to find its path
    */
   function getPathToTree(instance, tree) {
     const pathComponents = [];
-    let parenthood = instance._parenthoodMap.get(tree);
-    while (parenthood && parenthood.key) {
+    let treeMetadata = instance._treeMetadataMap.get(tree);
+    while (treeMetadata && treeMetadata.keyInParent) {
       // because we're walking up-tree, we need to use the array as a stack
-      pathComponents.unshift(parenthood.key);
-      parenthood = instance._parenthoodMap.get(parenthood.parent);
+      pathComponents.unshift(treeMetadata.keyInParent);
+      treeMetadata = instance._treeMetadataMap.get(treeMetadata.parent);
     }
     if (pathComponents.length) {
       const path = pathComponents.join('/');
       return '/' + path;
     }
     return '';
   }
+  /**
+   * A callback to be used as the proxy get trap callback.
+   * It allows to check get the tree metadata from within the tree. This allows to check if the object is proxified by this
+   * instance of JSONPatcherProxy, even when all you have is a reference that is someone else's proxy
+   * @param {JSONPatcherProxy} instance JSONPatcherProxy instance
+   * @param {Object} tree the affected object
+   * @param {String} key the effect property's name
+   */
+  function trapForGet(instance, tree, key) {
```

Comment:
It would be nice to add a test to benchmarks to check how much we slowdown regular work with proxified object. As I suspect adding a trap for getting any property in entire tree could affect performance.

Reply:
Good idea, I will add such test.